File: finder4j-backtrack/src/test/java/com/github/k0zka/finder4j/backtrack/listener/LoggerSolutionListenerTest.java
package com.github.k0zka.finder4j.backtrack.listener;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.runners.MockitoJUnitRunner;

import com.github.k0zka.finder4j.backtrack.State;
import com.github.k0zka.finder4j.backtrack.Step;

@RunWith(MockitoJUnitRunner.class)
public class LoggerSolutionListenerTest {
	@Mock
	State state;

	@Test
	public void onSolution() {
		new LoggerSolutionListener<State, Step<State>>().onSolution(state);
		// verify that no error thrown: done
	}
}

File: finder4j-backtrack/src/test/kotlin/com/github/k0zka/finder4j/backtrack/listener/LoggerSolutionListenerTest.kt
package com.github.k0zka.finder4j.backtrack.listener

import com.github.k0zka.finder4j.backtrack.State
import com.nhaarman.mockito_kotlin.mock
import org.junit.Test

class LoggerSolutionListenerTest {
	private val state: State = mock()

	@Test
	fun onSolution() {
		LoggerSolutionListener<State>().onSolution(state)
		// verify that no error thrown: done
	}
}
